ghcide/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -118,13 +119,14 @@ runLanguageServer recorder options inH outH defaultConfig parseConfig onConfigCh
     let lspCologAction :: MonadIO m2 => Colog.LogAction m2 (Colog.WithSeverity LspServerLog)
         lspCologAction = toCologActionWithPrio (cmapWithPrio LogLspServer recorder)
 
-    void $ untilMVar clientMsgVar $
-          void $ LSP.runServerWithHandles
+    untilMVar clientMsgVar $
+          LSP.runServerWithHandles
             lspCologAction
             lspCologAction
             inH
             outH
             serverDefinition
+          `finally` sequence_ onExit
```

Comment:
Comment with the explanation what `onExit` does, please :)

Perhaps introducing an intermediate variable for the `runServerWithHabdles` action?